holmes.go

```diff
@@ -703,12 +703,21 @@ func (h *Holmes) getCPUCore() (float64, error) {
 		return h.opts.cpuCore, nil
 	}
 
-	if h.opts.UseGoProcAsCPUCore {
-		return float64(runtime.GOMAXPROCS(-1)), nil
+	if h.opts.UseCGroup {
+		// If executed get cpuCore equal to 0 (most likely because no resource limit is set),
+		// Then only print out err, and finally return the number of CPU logic cores compatible.
+		cpuCore, err := getCGroupCPUCore()
+		if err != nil {
+			h.Warnf("[Holmes] get CGroup CPU core failed, CPU core: %v, error: %v", cpuCore, err)
```

Comment:
Does it loss the `return` statement?

Reply:
It’s not lost, it’s when the `getCGroupCPUCore` doesn’t find the value, return by `runtime.GOMAXPROCS(-1)` or `runtime.NumCPU()`.
So print the warning log, but let the service sample normally.

Reply:
1. `cpuCore` seems no need to log printout since it  must equal 0 while `getCGroupCPUCore` returns err
2. I think it's better to remind users that we take another way to obtain CPU core numbers while `getCGroupCPUCore` returns err. Maybe we can add some information on L711. What's your take on that?